xbanish.c

```diff
@@ -155,6 +156,16 @@ main(int argc, char *argv[])
 		else if (e.type == button_press_type ||
 		    e.type == button_release_type)
 			etype = ButtonRelease;
+		else if (e.type == device_change_type) {
+			DPRINTF(("device list changed, snooping again\n"));
+			if (snoop_xinput(DefaultRootWindow(dpy)) == 0) {
+				DPRINTF(("no XInput devices found, "
+				    "using legacy snooping"));
+				legacy = 1;
+				snoop_legacy(DefaultRootWindow(dpy));
+			}
```

Comment:
Maybe move this code into a `snoop_root` function and call it here, and from earlier in `main` where this code is duplicated.